R/load_data.R

```diff
@@ -1,7 +1,147 @@
-#' Load Raw Billions Indicator Data
+#' Load Billions indicator data
 #'
 #' `load_billion_data()` provides access to the raw inputs to the Billions stored
-#' within the World Health Organization's xMart4 database. By default, it filters
+#' within the World Health Data Hub or the World Health Organization's xMart4 database.
+#' By default, it filters the loaded data to take the most recently uploaded data
+#' for each indicator, country, and year. The user can also specify to take the
+#' latest data before a certain time period, or to extract all data from the database.
+#'
+#' This function requires that the user have the [xmart4](https://github.com/gpw13/xmart4)
+#' and [whdh](https://github.com/WorldHealthOrganization/whdh) packages installed
+#' and setup to access the data. For quetions about getting the relevant permissions,
+#' please contact <kanjim@who.int> or <messeillere@who.int>.
+#'
+#' @param data_type (string) The type of data to load.
+#' * `wrangled_data` (default): raw data that has been wrangled into a suitable
+#'   form for analysis.
+#' * `projected_data`:  data that has been fully projected to the target year but
+#'   has not yet been transformed or calculated upon.
+#' * `final_data`: the complete set of billions data with transformed values, contributions,
+#'   and all calculations available.
+#' @param billion (string) One of "all" (default), "hep", "hpop", or "uhc". If "all",
+#'   downloads all indicators for all three billions.
+#' @param ind_codes (character vector) The name of the indicator (or indicators) to load data for.
+#'   If `all`, downloads data for all indicators for a given billion. Ignored if
+#'   billion = "all".
+#' @param date_filter (string) Either "latest", or a single date string. The date
+#'   string needs to be in ISO6801 format, such as "1989-4-4" or "1988-06-21".
+#'   If a date is provided, the returned data frame is a snapshot of the data as
+#'   stored on that date (if no updates were made on the given date, then the most
+#'   recent update before that date is used).
+#' @param na_rm (logical) Specifies whether to filter the data to only rows
+#'   where `value` is not missing. Defaults to `TRUE`.
+#' @param sandbox (logical) Whether or not to use the sandbox folders in the data lake.
+#' @param silent (logical) Specifies whether to show authentication messages and
+#'   a download progress bar. Defaults to `TRUE`.
+#' @param data_source (string) Ether "whdh" or "xmart". Indicates where to download
+#'   the data from
+#' @param ... Additional arguments passed to `xmart4::xmart4_table()`. Use if
+#'   you need to provide additional token specifications for Azure authentication.
+#'
+#' @return A data frame.
+#'
+#' @export
+#'
+load_billion_data <- function(data_type = c("wrangled_data", "projected_data", "final_data"),
```

Comment:
I know you don't like this, but load_billion_data needs to support legacy use. You could add the old table names (raw_data, proj/unproj_data, etc.)  to data_type, then force the use of xmart for those and return a warning informing the user that it will be discontinued.

This makes me wonder if we should start to use the [lifecycle package](https://lifecycle.r-lib.org/) too

Reply:
Moved into a separate issue: #45. Resolving for now.